packages/@aws-cdk/aws-apigateway/test/integ.stepfunctions-api.ts

```diff
@@ -0,0 +1,41 @@
+import * as sfn from '@aws-cdk/aws-stepfunctions';
+import * as cdk from '@aws-cdk/core';
+import { Construct } from 'constructs';
+import * as apigw from '../lib';
+
+/**
+ * Stack verification steps:
+ * * `curl -X POST 'https://<api-id>.execute-api.<region>.amazonaws.com/prod' \
+ * * -d '{"key":"Hello"}' -H 'Content-Type: application/json'`
+ * The above should return a "Hello" response
+ */
+
+class StepFunctionsRestApiDeploymentStack extends cdk.Stack {
+  constructor(scope: Construct) {
+    super(scope, 'StepFunctionsRestApiDeploymentStack');
+
```

Comment:
This is only testing the happy case. Can we include 4xx and 5xx error responses in the integ test? This will help ensure future changes to those configuration don't regress.

Reply:
Integration test deploys the stack, verifies the resources and then cleans it up.

Reply:
picking up the pull-request from here. I discussed with Saqib today and unless a new Lambda is made and invoked, there's no way to invoke the URL to validate 400s/500s requests. Is it possible to do that from the CDK package? Saqib added integration tests that validate whether the generated stack is correctly deployed, like the LambdaRestApi construct.